website/pages/docs/internals/limits.mdx

```diff
@@ -0,0 +1,271 @@
+---
+layout: docs
+page_title: Limits and Maximums
+sidebar_title: Limits and Maximums
+description: Learn about the maximum number of objects within Vault.
+---
+
+# Vault Limits and Maximums
+
+Vault imposes fixed upper limits on the size of certain fields and
+objects, and configurable limits on others.  Vault also has upper
+bounds that are a consequence of its underlying storage. This page
+attempts to collect these limits, to assist in planning Vault
+deployments.
+
+In some cases, the system will show performance problems in advance of
+the absolute limits being reached.
+
+## Storage-Related limits
+
+### Storage entry size
+
+The maximum size of an object written to a storage backend is determined
+by that backend.
+
+For the Consul storage backend, the default limit imposed by Consul is
+512 KiB.  This may be configured via Consul’s
+[`kv_max_value_size`](https://www.consul.io/docs/agent/options#kv_max_value_size) parameter, introduced in version 1.5.3.
+
+For the integrated storage backend, the default limit (introduced in
+Vault 1.5.0) is 1 MiB.  This may be configured via `max_entry_size` in
+the [storage stanza](/docs/configuration/storage/raft#max_entry_size).
+
+Many of the other limits within Vault derive from the maximum size of
+a storage entry, as described in the next sections. It is possible to
+recover from an error where a storage entry has reached its maximum
+size by reconfiguring Vault or Consul to a larger maximum storage
+entry.  However, using large storage entries also negatively affects
```

Comment:
Do we want to allude to this being an issue not just in terms of bigger writes to storage locally, but the fact that Raft doesn't like them?  e.g. from the raft repo:

    // Increasing beyond this risks RPC IO taking too long and preventing
    // timely heartbeat signals which are sent in serial in current transports,
    // potentially causing leadership instability.

Reply:
My vote would be to definitely call this out. We don't want customers thinking that it's an "easy fix" to just increase the max storage entry size and the more technical justification we can provide, the better.

Reply:
Added.